tests/lib/cli-engine/cli-engine.js

```diff
@@ -3121,18 +3169,29 @@ describe("CLIEngine", () => {
                 `
             };
 
-            it("should lint only JavaScript blocks if '--ext' was not given.", () => {
-                CLIEngine = defineCLIEngineWithInMemoryFileSystem({
-                    cwd: () => root,
+            let cleanup;
+
+            beforeEach(() => {
+                cleanup = () => {};
+            });
+
+            afterEach(() => cleanup());
+
+            it("should lint only JavaScript blocks if '--ext' was not given.", async () => {
+                const teardown = createCustomTeardown({
+                    cwd: root,
                     files: {
                         ...commonFiles,
-                        ".eslintrc.json": JSON.stringify({
+                        ".eslintrc.json": {
                             plugins: ["markdown", "html"],
                             rules: { semi: "error" }
-                        })
+                        }
                     }
-                }).CLIEngine;
-                engine = new CLIEngine({ cwd: root });
+                });
+
+                cleanup = teardown.cleanup;
+                await teardown.prepare();
```

Comment:
Should we switch the order like this (to set `cleanup` before doing `prepare()`) in other tests as well, for consistency (there are 19 tests with this pattern, can be found by searching for `await teardown.prepare()`)

Reply:
I don't think this is important enough to spend time fixing. Let's focus on higher level issues and keep in mind this file will be going away once we remove `CLIEngine`.